qgis2web/maindialog.py

```diff
@@ -353,11 +345,10 @@ def populate_layers_and_groups(self, dlg):
 
         for tree_layer in tree_layers:
             layer = tree_layer.layer()
-            if (layer.type() != QgsMapLayer.PluginLayer and
-                    layer.customProperty("ol_layer_type") is None):
+            if (layer.type() != QgsMapLayer.PluginLayer and (layer.type() != QgsMapLayer.VectorLayer or layer.wkbType() != QgsWkbTypes.NoGeometry) and layer.customProperty("ol_layer_type") is None):
                 try:
-                    if layer.type() == QgsMapLayer.VectorLayer:
-                        testDump = layer.renderer().dump()
+                    # if layer.type() == QgsMapLayer.VectorLayer:
+                    #    testDump = layer.renderer().dump()
```

Comment:
There was a reason for this, long ago, but I'd have to dig out the commit to see what issue it solved. I'll check blame.

Reply:
maybe for debugging purposes - testDump is not actually used anywhere - and failed when layers without a renderer (NoGeometry) still got here.

Reply:
I think this was simply to test whether the layer could be queried for its renderer, and to skip the layer if not.

Reply:
Yes: https://github.com/tomchadwin/qgis2web/commit/390431c974b6b46fce53598cc0a6fd7fce6f6783#diff-3a7deece77edeafc97a5df780ca99df0R625. You are now testing explicitly for layers without geometry, so this should be fine, unless it was catching other undetected cases where the layer should be skipped.